### PR TITLE
realm-name: Add support for realm urls having '/api' at the end.

### PR DESCRIPTION
**Pull request by aero31aero**
Relevant conversation on:
1. https://chat.zulip.org/#narrow/stream/bots/topic/zulip-js.20api.20bindings
2. https://chat.zulip.org/#narrow/stream/test.20here/topic/github-issues

### CODE REVIEW

File: test/zuliprc [hunk 1]
```diff
@@ -1,4 +1,4 @@
 [api]
-email=cordelia@zulip.com
-key=wlueAg7cQXqKpUgIaPP3dmF4vibZXal7
-site=http://localhost:9991
+email=api-test-bot@chat.zulip.org
+key=K1w7xuUlCorRXjDCz2EkCC4IXMbdYOpK
+site=https://chat.zulip.org
```
Comment (timabbott) [inline test/zuliprc]: We should test against a Zulip development environment, not a production site.

Comment (aero31aero) [inline test/zuliprc]: How do I do that when running the tests in Travis? I was confused so did this.

Comment (timabbott) [inline test/zuliprc]: To do that would be significant setup work -- we'd need to basically run the main Zulip server project's development environment provisioning.

I guess the way to make that easy would be to merge this project into zulip.git.

Do the tests fail if you just include the changes actually related to the feature in this PR?

Comment (aero31aero) [inline test/zuliprc]: Tests were already failing for several builds, so my commit didn't make any difference to that. First commit is manually tested and merge-able.

**Comment by aero31aero** [comment]
@timabbott removed the second commit, which made the tests pass(sort of forcefully). This should be merge-able. I'm looking into more ways of testing this in the meantime. Local tests are passing for the code I added.

**Comment by timabbott** [comment]
@arpith is it convenient for you to post a release on NPM?

**Comment by arpith** [comment]
Done 👍🏼

**Pull request merged**